code/modules/antagonists/roundstart_special/undercover/undercover.dm

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////
+////// Special Role Datum///////
+////////////////////////////////
+
+/datum/special_role/undercover
+	probability = 65			//The probability of any spawning at all
+	proportion = 0.05			//The prbability per person of rolling it (5% is (5 in 100) (1 in 20))
+	max_amount = 4				//The maximum amount
+	role_name = "Undercover Agent"
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Head of Personnel", "Chief Medical Officer", "Chief Engineer", "Research Director", "Captain", "Brig Physician", "Clown")
+	attached_antag_datum = /datum/antagonist/special/undercover
+
+////////////////////////////////
+//////  Antagonist Datum ///////
+////////////////////////////////
+
+/datum/antagonist/special/undercover
+	name = "Ex-security agent"
+	roundend_category = "Special Roles"
+	antag_moodlet = /datum/mood_event/determined
+
+/datum/antagonist/special/undercover/greet()
+	to_chat(owner, "<span class='userdanger'>You are an ex-security agent.</span>")
+	to_chat(owner, "<b>Due to your loyality to nanotrasen in the past, you have been granted with a weapon permit.</b>")
+	to_chat(owner, "<b>Additionally nanotrasen has authorised you to have a disabler for personal defense.</b>")
+	to_chat(owner, "<b>You are not a member of security, and shouldn't hunt criminals, but may use your weapon for self defense.</b>")
+	to_chat(owner, "<span class='boldannounce'>Do NOT commit traitorous acts in persuit of your objectives.</span>")
+
+/datum/antagonist/special/undercover/admin_add(datum/mind/new_owner, mob/admin)
+	. = ..()
+	var/mob/living/carbon/C = new_owner.current
+	if(!istype(C))
+		to_chat(admin, "You can only turn carbons into an ex-security agent.")
+		return
+	message_admins("[key_name_admin(admin)] made [key_name_admin(new_owner)] into an ex-security agent.")
+	log_admin("[key_name(admin)] made [key_name(new_owner)] into [name].")
+
+/datum/antagonist/special/undercover/forge_objectives()
+	var/datum/objective/saveshuttle/chosen_objective = new
+	chosen_objective.generate_people_goal()
+	objectives += chosen_objective
+
+	if(owner.assigned_role in GLOB.engineering_positions)
+		var/datum/objective/protect_sm/objective = new
+		if(objective.get_target())
+			objective.update_explanation_text()
+			objectives += objective
+
+	owner.announce_objectives()
+
+/datum/antagonist/special/undercover/equip()
+	if(!owner)
+		return
+
+	var/mob/living/carbon/H = owner.current
+	if(!ishuman(H) && !ismonkey(H))
+		return
+
+	var/obj/item/gun/energy/disabler/T = new(H)
+	var/obj/item/restraints/handcuffs/cable/zipties/T2 = new(H)
+	var/list/slots = list (
+		"backpack" = SLOT_IN_BACKPACK,
+		"left pocket" = SLOT_L_STORE,
+		"right pocket" = SLOT_R_STORE
+	)
+	var/where = H.equip_in_one_of_slots(T, slots)
+	H.equip_in_one_of_slots(T2, slots)
+	if (!where)
+		to_chat(owner, "<span class='warning'>You lost your weapon on the way here! You should be more careful next time.</span>")
```

Comment:
🤢

Reply:
Goes in active hand if not in backpack somehow

Reply:
No it doesn't, it'll qdel on fail by default. And even if you disable qdelling, you should check if it returns null and just `put_in_hands()` or spawn it on their turf.

Reply:
I changed the code, so it goes into the hand. I'll add something to spawn it on the turf too

Reply:
put_in_hands() should drop onto the turf by default if it fails

Reply:
This is the current code (Reviewed one is outdated)

```
	var/where = H.equip_in_one_of_slots(T, slots)
	H.equip_in_one_of_slots(T2, slots)
	if (!where)
		if(!H.put_in_hands(T))
			to_chat(owner, "<span class='warning'>Your weapon has been placed on the floor.</span>")
```